packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -134,57 +138,121 @@ export class AssetsContractController extends BaseController<
    * @property provider - Provider used to create a new underlying Web3 instance
    */
   set provider(provider: any) {
-    this._provider = new Web3Provider(provider);
-    this.erc721Standard = new ERC721Standard(this._provider);
-    this.erc1155Standard = new ERC1155Standard(this._provider);
-    this.erc20Standard = new ERC20Standard(this._provider);
+    this._provider = provider;
   }
 
   get provider() {
     throw new Error('Property only used for setting');
   }
 
+  /**
+   * Get the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID.
+   * @returns Web3Provider instance.
+   */
+  getProvider(networkClientId?: NetworkClientId): Web3Provider {
+    const provider = networkClientId
+      ? this.getNetworkClientById(networkClientId).provider
+      : this._provider;
+
+    if (provider === undefined) {
+      throw new Error(MISSING_PROVIDER_ERROR);
+    }
+
+    return new Web3Provider(provider);
+  }
+
+  /**
+   * Get the relevant chain ID.
+   *
+   * @param networkClientId - Network Client ID used to get the provider.
+   * @returns Hex chain ID.
+   */
+  getChainId(networkClientId?: NetworkClientId): Hex {
+    return networkClientId
+      ? this.getNetworkClientById(networkClientId).configuration.chainId
+      : this.config.chainId;
+  }
+
+  /**
+   * Get a ERC20Standard instance using the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID used to get the provider.
+   * @returns ERC20Standard instance.
+   */
+  getERC20Standard(networkClientId?: NetworkClientId): ERC20Standard {
+    const provider = this.getProvider(networkClientId);
+    return new ERC20Standard(provider);
+  }
+
+  /**
+   * Get a ERC721Standard instance using the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID used to get the provider.
+   * @returns ERC721Standard instance.
+   */
+  getERC721Standard(networkClientId?: NetworkClientId): ERC721Standard {
+    const provider = this.getProvider(networkClientId);
+    return new ERC721Standard(provider);
+  }
+
+  /**
+   * Get a ERC1155Standard instance using the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID used to get the provider.
+   * @returns ERC1155Standard instance.
+   */
+  getERC1155Standard(networkClientId?: NetworkClientId): ERC1155Standard {
+    const provider = this.getProvider(networkClientId);
+    return new ERC1155Standard(provider);
+  }
```

Comment:
I don't think it should be the token standards class' concern  that the idea of a `networkClientId` exists and that it needs to get a provider for it. If we did want to pass `networkClientId` and fetch the provider in the token standard constructor, we would also have to provide `getNetworkClientById` so that it has a way to get the provider. This doesn't seem to buy us much in my opinion

Reply:
yeah you might be right. This is definitely a convo for another day. I think the `AssetsContractController` itself is needs a broader rethink. Its not clear why we need/want it to instantiate the standards classes and forward the calls for single method calls. But this is a much broader discussion.